packages/samples/vuejs/README.md

```diff
@@ -1,19 +1,231 @@
+
+```json
+scripts: {
+  ...,
+  "dev": "npm run copy:assets && npm run serve",
+  "copy:assets": "ncp node_modules/@coveo/atomic/dist/atomic/assets public/assets && ncp node_modules/@coveo/atomic/dist/atomic/lang public/lang"
+}
+```
+
+> The `dev` script assumes you are using the default `serve` script and `public/assets` path for running your application. Don't forget to adjust if needed.
+
+### Update your entry file
+
+The atomic package allows you to customise your components by defining CSS variables in your own stylesheet - you can read about that [here](https://docs.coveo.com/en/atomic/latest/usage/themes-and-visual-customization/).
+
+It also offers you a default theme that you can use as is or build upon. Those styles live in `@coveo/atomic/dist/atomic/themes/coveo.css` and you can import them in your entry js file like below.
+
+
+```ts
+import '@coveo/atomic/dist/atomic/themes/coveo.css'; // optional!
+import {applyPolyfills, defineCustomElements} from '@coveo/atomic/loader';
+import {createApp} from 'vue';
+import App from './App.vue';
+
+// Bind the custom elements to the window object [https://stenciljs.com/docs/vue]
+applyPolyfills().then(() => {
+  defineCustomElements(window);
+});
+
+createApp(App).mount('#app');
 ```
 
-### Compiles and minifies for production
+## Using the CDN
+
+Follow the instructions [here](https://docs.coveo.com/en/atomic/latest/usage/#cdn) to add the CDN script to your entry html file. The script will initialize your components, avoiding the need to manually do it like in the npm package usage above. You will also not need to copy any assets over.
+
+## Initializing your search interface
+
+After you render your search interface, make sure to initialize it
+
+```js
+import {onMounted} from 'vue';
+
+async function initInterface() {
+  await customElements.whenDefined('atomic-search-interface');
+  const searchInterface = document.querySelector(
+    'atomic-search-interface'
+  ) as HTMLAtomicSearchInterfaceElement;
+
+  // Initialization
+  await searchInterface.initialize({
+    accessToken: '<YOUR-TOKEN>',
+    organizationId: '<YOUR-ORGANIZATION-ID>',
+  });
+
+  // Trigger a first search
+  searchInterface.executeFirstSearch();
+}
+
+onMounted(initInterface);
 ```
-npm run build
+
+### Using `<atomic-result-template>`
+
+As you are aware, Vue treats the `<template>` tag as a container for Vue components and therefore doesn't render it to the page. This poses an obstacle for working with the `<atomic-result-template>`, which expects a native `<template>` that it will then use to create each result element. In order for this to work, your result templates will need to live in an html file, like in this example's `src/templates/result-template.html` and then imported and used like below
```

Comment:
```suggestion
As you are aware, Vue treats the `<template>` tag as a container for Vue components and therefore doesn't render it to the page. This poses an obstacle for working with the `<atomic-result-template>`, which expects a native `<template>` that it will then use to create each result element. In order for this to work, your result templates will need to live in an HTML file (like in this example's `src/templates/result-template.html` file) and then imported and used like below
```